test-rt/suite-unit/src/bin_einsum.rs

```diff
@@ -0,0 +1,396 @@
+                    if params.force_max_one_iter_axis { 0..2usize } else { 0..3usize };
+                n_axes_range.prop_flat_map(move |supp_n_axes| {
+                    let remaining = remaining - supp_n_axes;
+                    let iter_axes_range = if remaining < iter_axes_range.end {
+                        0..(remaining + 1)
+                    } else {
+                        iter_axes_range.clone()
+                    };
+                    iter_axes_range.clone().prop_flat_map(move |iter_axes| {
+                        let remaining = remaining - iter_axes;
+                        let trivial_m_n_axes_range =
+                            if params.no_trivial_axes { 0..1usize } else { 0..2usize };
+                        let trivial_m_axes_range = if remaining < trivial_m_n_axes_range.end {
+                            0..(remaining + 1)
+                        } else {
+                            trivial_m_n_axes_range.clone()
+                        };
+                        trivial_m_axes_range.clone().prop_flat_map(move |trivial_m_axes| {
+                            let remaining = remaining - trivial_m_axes;
+                            let trivial_n_axes_range = if remaining < trivial_m_n_axes_range.end {
+                                0..(remaining + 1)
+                            } else {
+                                trivial_m_n_axes_range.clone()
+                            };
+                            trivial_n_axes_range.clone().prop_flat_map(move |trivial_n_axes| {
+                                Just((
+                                    supp_m_axes,
+                                    supp_n_axes,
+                                    iter_axes,
+                                    trivial_m_axes,
+                                    trivial_n_axes,
+                                ))
+                            })
+                        })
+                    })
+                })
+            })
+            .prop_map(|(supp_m_axes, supp_n_axes, iter_axes, trivial_m_axes, trivial_n_axes)| {
+                dbg!(supp_m_axes, supp_n_axes, iter_axes, trivial_m_axes, trivial_n_axes);
+                let m_axes: String = ('b'..).take(supp_m_axes).collect();
+                let trivial_m_axes: String = ('m'..).take(trivial_m_axes).collect();
+                let trivial_n_axes: String = ('p'..).take(trivial_n_axes).collect();
+                let n_axes: String = ('h'..).take(supp_n_axes).collect();
+                let iter_axes: String = ('w'..).take(iter_axes).collect();
+                let a_axes: Vec<char> =
+                    (m_axes.clone() + "a" + &trivial_m_axes + &iter_axes + "k").chars().collect();
+                let b_axes: Vec<char> =
+                    (n_axes.clone() + "g" + &trivial_n_axes + &iter_axes + "k").chars().collect();
+                let c_axes: Vec<char> =
+                    (m_axes + &n_axes + "ag" + &trivial_m_axes + &trivial_n_axes + &iter_axes)
```

Comment:
for our sanity, is it possible to re-arrange letters affectation so that "a" becomes "m" and "g" besomes "n" ? :)